contribs/gmf/src/directives/editfeature.js

```diff
@@ -0,0 +1,446 @@
+
+  /**
+   * @type {ngeo.ToolActivateMgr}
+   * @private
+   */
+  this.ngeoToolActivateMgr_ = ngeoToolActivateMgr;
+
+  /**
+   * @type {number}
+   * @private
+   */
+  this.pixelBuffer_ = 10;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.createActive = false;
+
+  /**
+   * @type {ngeo.ToolActivate}
+   * @export
+   */
+  this.createToolActivate = new ngeo.ToolActivate(this, 'createActive');
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.mapSelectActive = true;
+
+  $scope.$watch(
+    function() {
+      return this.mapSelectActive;
+    }.bind(this),
+    this.handleMapSelectActiveChange_.bind(this)
+  );
+
+  /**
+   * @type {ngeo.ToolActivate}
+   * @export
+   */
+  this.mapSelectToolActivate = new ngeo.ToolActivate(this, 'mapSelectActive');
+
+  /**
+   * @type {?ol.Feature}
+   * @export
+   */
+  this.feature = null;
+
+  /**
+   * @type {ol.Collection}
+   * @export
+   */
+  this.features = this.vectorLayer.getSource().getFeaturesCollection();
+
+  /**
+   * @type {?Array.<ngeox.Attribute>}
+   * @export
+   */
+  this.attributes = null;
+
+  /**
+   * @type {?string}
+   * @export
+   */
+  this.geomType = null;
+
+  gmfXSDAttributes.getAttributes(this.layer.id).then(
+    this.setAttributes_.bind(this));
+
+  var uid = goog.getUid(this);
+  this.eventHelper_.addListenerKey(
+    uid,
+    ol.events.listen(
+      this.features,
+      ol.CollectionEventType.ADD,
+      this.handleFeatureAdd_,
+      this
+    )
+  );
+
+  $scope.$on('$destroy', this.handleDestroy_.bind(this));
+
+  this.toggle_(true);
+};
+
+
+/**
+ * Save the currently selected feature modifications.
+ * @export
+ */
+gmf.EditfeatureController.prototype.save = function() {
+  var feature = this.feature;
+  var id = feature.getId();
+
+  if (id) {
+    this.editFeatureService_.updateFeature(
+      this.layer.id,
+      feature
+    ).then(
+      this.handleEditFeature_.bind(this)
+    );
+  } else {
+    this.editFeatureService_.insertFeatures(
+      this.layer.id,
+      [feature]
+    ).then(
+      this.handleEditFeature_.bind(this)
+    );
+  }
+};
+
+
+/**
+ * @export
+ */
+gmf.EditfeatureController.prototype.cancel = function() {
+  this.feature = null;
+  this.features.clear();
+};
+
+
+/**
+ * @export
+ */
+gmf.EditfeatureController.prototype.delete = function() {
+  // (1) Launch request
+  this.editFeatureService_.deleteFeature(
+    this.layer.id,
+    this.feature
+  ).then(
+    this.handleDeleteFeature_.bind(this)
+  );
+
+  // (2) Reset selected feature
+  this.feature = null;
+  this.features.clear();
+
+};
+
+
+/**
+ * Called after an insert, update or delete request.
+ * @param {angular.$http.Response} resp Ajax response.
+ * @private
+ */
+gmf.EditfeatureController.prototype.handleEditFeature_ = function(resp) {
+  var features = new ol.format.GeoJSON().readFeatures(resp.data);
+  if (features.length) {
+    var feature = features[0];
+    var id = feature.getId();
+    this.feature.setId(id);
```

Comment:
I'll make the change.

I usually make the code as "human-readable" as possible.  I.e. it's easier to read the code with 3 lines instead of this one liner.  I do so because, in the end, everything gets simplified and minimized by the Compiler anyway.  What I mean is: the change doesn't really matter in the end.  Thoughts ?